addons/product_expiry/product_expiry.py

```diff
@@ -105,7 +105,7 @@ def _get_quants(self, cr, uid, ids, context=None):
 
     def apply_removal_strategy(self, cr, uid, location, product, qty, domain, removal_strategy, context=None):
         if removal_strategy == 'fefo':
-            order = 'removal_date, in_date, id'
+            order = 'removal_date, location_id, package_id, lot_id, in_date, id'
```

Comment:
@clonedagain, yes i agree because on fefo removal strategy, material requirements are serviced in the order of items with the earlier date of consumption regardless of the date of entry or acquisition.